infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/Orchestrator.java

```diff
@@ -234,7 +241,12 @@ void run(@Nonnull IWorkflow workflow, int actionRetry) {
 
                 log.debug("run: Started action {} for workflow {}", action.getName(), workflow.getId());
                 long actionStart = System.currentTimeMillis();
-                action.execute(rt, actionRetry);
+                if(action instanceof RestoreAction){
+                    ((RestoreAction) action).execute(rt, streamLog, actionRetry);
```

Comment:
This looks like a hack. Can't we pass additional parameters to the actions on execution?

Reply:
Ok, I'll figure out how to make it better